src/main/java/frc/trigon/robot/subsystems/roller/simulationroller/SimulationRollerIO.java

```diff
@@ -0,0 +1,75 @@
+package frc.trigon.robot.subsystems.roller.simulationroller;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.roller.RollerIO;
+import frc.trigon.robot.subsystems.roller.RollerInputsAutoLogged;
+import frc.trigon.robot.utilities.Conversions;
+
+public class SimulationRollerIO extends RollerIO {
+    private final SingleJointedArmSim angleSimulation = SimulationRollerConstants.ANGLE_MOTOR;
+    private final DCMotorSim collectorSimulation = SimulationRollerConstants.COLLECTOR_MOTOR;
+
+    private double angleMotorVoltage, collectorMotorVoltage;
+
+    @Override
+    protected void updateInputs(RollerInputsAutoLogged inputs) {
+        angleSimulation.update(RobotConstants.PERIODIC_TIME_SECONDS);
+        collectorSimulation.update(RobotConstants.PERIODIC_TIME_SECONDS);
+
+        inputs.angleMotorVoltage = angleMotorVoltage;
+        inputs.angleMotorCurrent = angleSimulation.getCurrentDrawAmps();
+        inputs.collectionMotorVoltage = collectorMotorVoltage;
+        inputs.collectionMotorCurrent = collectorSimulation.getCurrentDrawAmps();
+        inputs.forwardLimitSwitch = angleSimulation.hasHitUpperLimit();
+        inputs.backwardLimitSwitch = angleSimulation.hasHitLowerLimit();
+    }
+
+    @Override
+    protected void setAngleMotorPower(double power) {
+        setAngleMotorPower(power);
+    }
+
+    @Override
+    protected void setCollectionMotorPower(double power) {
+        setCollectionMotorPower(power);
+    }
```

Comment:
You're calling the function inside the function. Call the set voltage instead and make sure to change the power to voltage